src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -282,8 +324,14 @@ private static String getClassNameFromQualifiedName(String qualifiedName) {
         /** Does class have private ctors. */
         private boolean withPrivateCtor;
 
-        /** Does class have nested subclass. */
-        private boolean withNestedSubclass;
+        /** Does class have subclasses. */
+        private boolean declaredAsParent;
+
+        /** Level of nesting of current class. */
+        private int depthLevel;
+
+        /** Node of AST that store class pkg token. */
+        private DetailAST astNode;
```

Comment:
It is not clear from name which node `ast` is referring to. Comment is slightly more confusion as a `class` doesn't have a package.
Please rename this field to something more meaningful and update description.